giftless/transfer/basic_streaming.py

```diff
@@ -80,9 +81,13 @@ def get(self, organization, repo, oid):
         """
         self._check_authorization(organization, repo, Permission.READ, oid=oid)
         path = os.path.join(organization, repo)
+
+        filename = request.args.get('filename')
+        headers = {'Content-Disposition': f'attachment; filename={filename}'} if filename else None
```

Comment:
I'm not sure, but we may want to escape `{filename}` here somehow to make it safe; Using it without some escaping or filtering in a response header *might* turn out to be a security risk. A good place to start reading about it is https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition but I would do some more searching. If you're not sure how to escape / encode the file name, let me know and I'll do some research.

I'm pretty sure that the file name should be in double quotes though.

Reply:
This is the only issue I remain unsure how to resolve.  I've done a little research but struggling to find much to latch onto.  I find some people using the escaped string here,  as in  `filename.encode('unicode_escape')` but that actually changes the desired name of the file you are returned. Can you give any further pointers as to the sort of thing you think is needed?

Reply:
It is indeed tricky. There is [RFC-5987](https://tools.ietf.org/html/rfc5987) which may hold the answer but it is pretty complex and may not be supported by all browsers. I'd take a naive approach of URL-encoding the file name (probably using `urllib.parse.quote`) but we have to first see how this is handled by a couple of different browsers. e.g. send a file name that has some special characters in it like new-line and see what happens when Chrome / FF / Edge / Safari tries to download it. I'm not sure you have time for such testing. 

Another approach, probably simpler and will work for 99% of cases, is not to escape but filter - replace everything that isn't "safe" (e.g. quotes, new lines, non-printing characters etc., need to figure out what characters are "safe") with something like a dot or a dash. 

Then put double quotes all around it.

Reply:
I've created a util function into which which ever strategy we think best for making filenames safe for HTTP headers.  At the moment there is just a simple whitelist approach implemented (along the lines of what you mention) - it just strips problematic chars from the filename.  It seemed the simplest and most transparent solution on the table at the moment.

Reply:
Something else worth considering to drop in here is the core ckan solution to this problem - a function "munge_filename" which can be found here: https://github.com/ckan/ckan/blob/6a77d2895e219e13fb38dd7f3ff33687f4609505/ckan/lib/munge.py#L144